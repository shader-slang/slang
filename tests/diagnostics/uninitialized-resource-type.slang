//DIAGNOSTIC_TEST:SIMPLE: -target hlsl -DTEST_1
//DIAGNOSTIC_TEST:SIMPLE: -target hlsl -DTEST_2

SamplerState sampler;

struct Foo
{
    bool sample_bar = false;
#ifdef TEST_1
    Texture2D<float4> bar = {};
#elif defined(TEST_2)
    Texture2D<float4> bar[2];
#endif
};

struct Result
{
    float4 color = float4(0.0);
};

Result process(in Foo foo)
{
    Result result = {};

    if (foo.sample_bar) {
#ifdef TEST_1
        result.color = foo.bar.Sample(sampler, float2(0.0, 0.0));
#elif defined(TEST_2)
        result.color = foo.bar[0].Sample(sampler, float2(0.0, 0.0));
#endif
    }

    return result;
}

[shader("compute")]
[numthreads(8, 8, 1)]
float4 cs_main(uint3 thread_id : SV_DispatchThreadID) {
    Foo foo;
    const let result = process(foo);
    return result.color;
}
