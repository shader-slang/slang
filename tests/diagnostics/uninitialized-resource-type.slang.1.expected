result code = -1
standard error = {
tests/diagnostics/uninitialized-resource-type.slang(40): warning 41016: use of uninitialized variable 'foo'
    const let result = process(foo);
                              ^
tests/diagnostics/uninitialized-resource-type.slang(40): error 56003: use of uninitialized opaque handle 'array<Texture2D,2>'.
    const let result = process(foo);
                              ^
}
